README.md

```diff
@@ -1,208 +1,262 @@
+This certificate needs to be converted into `.p12` format. If you don’t know how to do it, please refer to [this page]  (https://developer.apple.com/library/ios/documentation/IDEs/Conceptual/AppDistributionGuide/MaintainingCertificates/MaintainingCertificates.html).
 
-<img src="assets/images/step5.png"/>
+### Step 6 - Configure APNs on your application
+
+For the last step you need to configure the APNs setting within your application. This can be done by logging into your application as a user with Administrator role and opening the Apple Administration page that was set up in step 7 of the Implementation Guide.
+
+For this purpose you need to:
+-	Add your Apple Push Notification service SSL certificate in `.p12` format
+-	Add server url and port. For sandbox it is `gateway.sandbox.push.apple.com:2195` and `gateway.push.apple.com:2195` for production.
+-	Add feedback url and port. For sandbox it is `feedback.sandbox.push.apple.com:2196` and `feedback.push.apple.com:2196` for production.
+
+## Setting up Google Cloud Messaging Server
+
+In order to send google push notifications from this module you need to have set up a google account with google cloud messaging enabled.
+To do so follow these steps to get registered for Google cloud messaging and enter the details into the Mendix screens.
+
+### Step 1 - Login to developers console
+
+Open up the Google [developers console] (https://console.developers.google.com) and login with your Google id.
+<img src="assets/images/gcm-step1.png"/>
+
+### Step 2 - Create project
+
+From the `Go to project` menu, click `Create a project` and fill in the project name and project ID for your application. Then click `Create`.
+
+<img src="assets/images/gcm-step2.png"/>
+
+### Step 3 - Enable Google Cloud Messaging
+
+Once created, click the link to the Google Cloud Messaging API (section `Mobile APIs`) and click the Enable button.
+<img src="assets/images/gcm-step3.png"/>
+
+### Step 4 - Adding credentials
+
+Click on the menu option credentials, located on the left hand side under the API Manager section.
+<img src="assets/images/gcm-step4.png"/>
+
+Click on the `Create credentials` button. You would want to choose an API key of type `Server`.
+
+<img src="assets/images/gcm-step4b.png"/>
+
+### Step 5 - Create API key
+
+Choose a name for your key and, optionally, restrict the IP addresses that can connect to the API.
+<img src="assets/images/gcm-step5.png"/>
+
+Then, press the `Create API key` button.
+
+<img src="assets/images/gcm-step5b.png"/>
+
+For the next step, you'll need to look up the **project number**. You can find it in your Google project's Project Information pane ( ⁝ utilities menu on the top right).
+
+<img src="assets/images/gcm-step6b.png"/>
+
+### Step 6 - Setup Mendix app
+
+Open up your application in Mendix and login as an Admin, so that you can see the menu option `Google admin`.
```

Comment:
'Open your Mendix application' 

We don't open an app **in** Mendix, and open up is kinda double :-)